Add friends likes

Eugenpak opened the pull request:
Спринт-11, ветка  Add friends likes

Eugenpak committed:
Редактировать пакет controller: FilmController, RestExceptionHandler, UserController.

Eugenpak committed:
Редактировать пакет model: Film, User.

Eugenpak committed:
Создать пакет service: FilmService, UserService.

Eugenpak committed:
Создать пакет storage: FilmStorage, UserStorage, InMemoryFilmStorage, InMemoryUserStorage.

Eugenpak committed:
Дополнить и добавить тесты в пакете controller: FilmControllerTest, UserControllerTest, FilmController2MockitoTest, UserController2MockitoTest.

Eugenpak committed:
Добавить тесты в пакете service: FilmServiceTest, UserServiceTest.

Eugenpak committed:
Редактировать прочее: pom.xml, application.properties.

Eugenpak committed:
Checkstyle Spr-11-v-1

Eugenpak committed:
Checkstyle Spr-11-v-2

Eugenpak committed:
Checkstyle Spr-11-v-3

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -15,73 +17,60 @@
+    }
+
+    @GetMapping(value = "/popular")
+    public Collection<Film> getPopularFilms(@RequestParam(required = false,defaultValue = "10") int count) {
```
defaultValue уже подразумевает, что параметра может и не быть. Поэтому вполне можно не писать  required = false

Eugenpak replied:
Исправил.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -0,0 +1,119 @@
+        }
+        if (!newUser.getEmail().contains("@")) {
+            log.error("ValidationException email");
+            throw new ValidationException("Этот имейл уже используется");
```
throw new ValidationException("Этот имейл уже используется");
.----------------

Здесь, скорее, неправильный формат, чем повторное использование.

Eugenpak replied:
Исправил.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -0,0 +1,119 @@
+        }
+        User oldUser = findUserById(newUser.getId());
+
+        if (newUser.getEmail() != null) {
```
Кстати, способ для проверки уникальности адреса: 
```java
    private boolean checkEmail(User user, Collection<User> userCollection) {
        return userCollection
                .stream()
                .filter(u -> u.getEmail().equals(user.getEmail())) // Фильтруем по email
                .anyMatch(u -> u != user); // Проверяем, что это не тот же самый пользователь
    }
```

Eugenpak replied:
Исправил. Хотя когда захотел сделать выброс исключение начались не проходить тесты в Postman.

avfyodorov replied:
Странно... нужно проверить ТЗ, может быть там нет такого требования и тесты не предусматривают выброс исключения в случае совпадения email

avfyodorov reviewed (commented):
Добрый день, Евгений!

Практически всё сделано, осталось чуть  поправить.

Eugenpak committed:
Исправить:
https://github.com/Eugenpak/java-filmorate/pull/2#discussion_r1975273694
https://github.com/Eugenpak/java-filmorate/pull/2#discussion_r1975282546
https://github.com/Eugenpak/java-filmorate/pull/2#discussion_r1975297071

avfyodorov reviewed (commented):
Добрый день, Евгений!

Замечаний нет.
Работа принята.
